lib/BuildSystem/BuildSystem.cpp

```diff
@@ -524,7 +524,18 @@ class CommandTask : public Task {
   }
 
   virtual void inputsAvailable(BuildEngine& engine) override {
-    command.inputsAvailable(getBuildSystem(engine).getCommandInterface(), this);
+    bool shouldSkip = !getBuildSystem(engine).getCommandInterface()
```

Comment:
BTW, this is off topic but I guess we should move all the `if (bsci.getDelegate().isCancelled()) {` checks here too, so we are consistent about where we implement the BuildSystem-layer special cases. What do you think?